src/main/java/elara/command/AddCommand.java

```diff
@@ -0,0 +1,98 @@
+package elara.command;
+
+import elara.storage.Storage;
+
+import elara.task.DeadlineException;
+import elara.task.DeadlineTask;
+import elara.task.EventException;
+import elara.task.EventTask;
+import elara.task.InvalidInputException;
+import elara.task.Task;
+import elara.task.TaskList;
+import elara.task.ToDoException;
+import elara.task.ToDoTask;
+
+import elara.ui.Ui;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class AddCommand implements Command {
+    private final String commandType;
+    private final String fullInput;
+    private final String details;
+
+    public AddCommand(String commandType, String fullInput) {
+        this.commandType = commandType;
+        this.fullInput = fullInput;
+        this.details = (fullInput.split(" ", 2).length > 1) ?  fullInput.split(" ", 2)[1] : null;
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws InvalidInputException {
+        Task newTask = null;
+
+        switch (commandType) {
+            case "todo":
+                if (details == null || details.isEmpty()) {
+                    throw new ToDoException();
+                }
+                newTask = new ToDoTask(details);
+                break;
+            case "deadline":
```

Comment:
Is the switch and case indentation here supposed to be the same according to the coding standard?

Reply:
fixed in later commits